be/src/runtime/tablets_channel.cpp

```diff
@@ -216,12 +216,12 @@ Status TabletsChannel::reduce_mem_usage(int64_t mem_limit) {
     // Sort the DeltaWriters by mem consumption in descend order.
     std::vector<DeltaWriter*> writers;
     for (auto& it : _tablet_writers) {
+        it.second->save_mem_consumption_snapshot();
```

Comment:
What's the core reason?

Reply:
> What's the core reason?

mem_consumption() returned value may be changed during sort period for same element, this does not meet comp requirements